modules/core/shared/src/test/scala/freestyle/tagless/TaglessTests.scala

```diff
@@ -34,6 +34,107 @@ import modules._
 
 class TaglessTests extends WordSpec with Matchers {
 
+  "the @tagless macro annotation should be accepted if it is applied to" when {
+
+    "a trait with at least one request" in {
+      "@tagless trait X { def bar(x:Int): FS[Int] }" should compile
+    }
+
+    "a trait with an F-Bound type param" in {
+      "@tagless trait FBound[F[_]] { def bar(x:Int): FS[Int] }" should compile
+    }
+
+    "an abstract class with at least one request" in {
+      "@tagless abstract class X { def bar(x:Int): FS[Int] }" should compile
+    }
+
+    "a trait with an abstact method of type FS" in {
+      "@tagless trait X { def f(a: Char) : FS[Int] }" should compile
+    }
+
+    "a trait with type parameters" ignore {
+      "@tagless trait X[A] { def ix(a: A) : FS[A] }" should compile
+    }
+
+    "a trait with some concrete non-FS members" ignore {
+      """@tagless trait X {
+        def x: FS[Int]
+        def y: Int = 5
+        val z: Int = 6
+      }""" should compile
+    }
+
+    "a trait with a request with multiple params" in {
+      "@tagless trait X { def f(a: Int, b: Int): FS[Int] }" should compile
+    }
+
+    "a trait with a currified request, with multiple params lists" ignore {
+      "@tagless trait X { def f(a: Int)(b: Int): FS[Int] }" should compile
+    }
+
+    "a trait with some implicit parameters, with many params lists" ignore {
+      "@tagless trait X { def f(a: Int)(implicit b: Int): FS[Int] }" should compile
+    }
+
+    "a trait with a request with no args" ignore {
+      "@tagless @debug trait X { def f: FS[Int] }" should compile
+    }
+
+    "a trait with a method with type parameters" in {
+      "@tagless trait WiX { def ix[A](a: A) : FS[A] }" should compile
+    }
+
+    "a trait with high bounded type parameters in the method" in {
+      "@tagless trait X { def ix[A <: Int](a: A) : FS[A] }" should compile
+    }
+
+    "a trait with lower bounded type parameters in the method" in {
+      "@tagless trait X { def ix[A >: Int](a: A) : FS[A] }" should compile
+    }
+
+    "a trait with different type parameters in the method" in {
+      "@tagless trait X { def ix[A <: Int, B, C >: Int](a: A, b: B, c: C) : FS[A] }" should compile
+    }
+
+    "a trait with high bounded type parameters and implicits in the method" in {
+      """
+        trait X[A]
+        @tagless trait Y { def ix[A <: Int : X](a: A) : FS[A] }
```

Comment:
Do we also support implicit arguments beside context bounds?

Reply:
There is a test with `implicit` arguments, yes: https://github.com/frees-io/freestyle/pull/491/files/91e17d1a7c0b09def6e51ff1be2a7dec3131c955#diff-f0ef9c02cc3b2899bf09ce3ea089ce66R75